sky/provision/gcp/instance_utils.py

```diff
@@ -1406,10 +1403,23 @@ def create_tpu_node(project_id: str, zone: str, tpu_node_config: Dict[str, str],
             _log_errors(provisioner_err.errors, e, zone)
             raise provisioner_err from e
 
+        provisioner_err.errors = [{
+            'code': 'UNKNOWN',
+            'domain': 'tpu',
+            'message': stderr
+        }]
+        _log_errors(provisioner_err.errors, e, zone)
+        raise provisioner_err from e
+
 
 def delete_tpu_node(project_id: str, zone: str, tpu_node_config: Dict[str,
                                                                       str]):
-    # Delete TPU node.
+    """Delete a TPU node with gcloud CLI.
+
+    This is used for both stopping and terminating a TPU node. It is ok to call
```

Comment:
In this case maybe best to name it `stop_or_delete_tpu_node()`. How does the CLI cmd determine when to stop and when to delete?

Reply:
Ahh, maybe I am not clear enough in the docstr. This function always delete the tpu accelerator, no matter it is stopping or terminating the cluster, because the host VM will be correctly stopped or terminated. Whenever we restart a stopped TPU node cluster, we will create a new TPU accelerator to attach to the host VM.